tree/ntuple/v7/src/RColumnElement.cxx

```diff
@@ -48,6 +48,31 @@ ROOT::Experimental::Detail::RColumnElementBase::Generate(EColumnType type) {
    return std::make_unique<RColumnElementBase>();
 }
 
+std::size_t ROOT::Experimental::Detail::RColumnElementBase::GetBitsOnStorage(EColumnType type) {
```

Comment:
I am confused.  Didn't we (also) need `GetBitsInMemory` which return `sizeof(underlying_type)`?

Reply:
The column type only pins down the size on storage for sure. A column element of a certain type can, at least potentially, loaded into differently sized memory slots.  For instance, a 32bit float could be loaded to memory into a page of floats or a page of doubles.  All possible combinations of [on-disk type, memory type] are written out as template specializations of `RColumnElement`

Reply:
Yes .. but I don't see (yet?) how the code we are talking about (which allocated then destruct a RColumnElement) would leverage any of that  (i.e. do anything but return the default mapping).  What am I missing?

Reply:
In [CommitSealedPage](https://github.com/root-project/root/pull/6763/files#diff-311b1306fdf38fbec1f6e0eeb3065c34b411f6d3f6f05614eb75cc02968f2a84R141) we allocate a concrete `RColumnElement` with its default memory mapping (we don't care about it, in fact) in order to call `GetBitsOnStorage()`.  That can now be replaced by the static `RColumnElementBase::GetBitsOnStorage(EColumnType)`.

In order to write the compression frame, we need both the compressed size and the uncompressed size.  The uncompressed size is the packed size of the page, which is (number of elements * bit on storage per element + 7) / 8.

Reply:
Ok but I still don't quite understand why it has to be done through code that will code that will essentially do 
```
auto x = new X;
auto res = x->GetSomeValue();
delete x;
return res;
```

Reply:
I see ... the code (at currently) just need GetBitsOnStorage.